apps/snitch_core/test/data/schema/payment/card_payment_test.exs

```diff
@@ -13,18 +26,100 @@ defmodule Snitch.Data.Schema.CardPaymentTest do
 
   describe "CardPayment records" do
     test "refer card type Payments (uniquely)", context do
-      %{ccd: card} = context
-      card_payment = CardPayment.changeset(%CardPayment{}, %{payment_id: card.id}, :create)
+      %{ccd: card_payment, user: user} = context
+
+      card_payment =
+        CardPayment.changeset(
+          %CardPayment{},
+          %{payment_id: card_payment.id, card: Map.put(@card, :user_id, user.id)},
+          :create
+        )
+
       assert {:error, %{errors: errors}} = Repo.insert(card_payment)
       assert errors == [payment_id: {"has already been taken", []}]
     end
 
     test "DON'T refer other Payments", context do
-      %{chk: check} = context
-      card_payment = CardPayment.changeset(%CardPayment{}, %{payment_id: check.id}, :create)
-      Repo.insert(card_payment)
+      %{chk: check, user: user} = context
+
+      card_payment =
+        CardPayment.changeset(
+          %CardPayment{},
+          %{payment_id: check.id, card: Map.put(@card, :user_id, user.id)},
+          :create
+        )
+
       assert {:error, %Ecto.Changeset{errors: errors}} = Repo.insert(card_payment)
       assert errors == [payment_id: {"does not refer a card payment", []}]
     end
+
+    test "with all good", context do
+      %{user: user, card_method: card_m, order: order} = context
+
+      payment_struct = %Payment{@payment | payment_method_id: card_m.id, order_id: order.id}
```

Comment:
Why do you have to make this struct? Isn't this identical to `context.ccd`?

Reply:
No it is not identical to context.ccd, because it returning the payment.id (ccd.id) which is already taken.

Reply:
You are mistaken, please see [line 135 of `factory.ex`](https://github.com/aviabird/snitch/blob/card_schema_init/apps/snitch_core/test/support/factory/factory.ex#L135) and you'll see that `context.ccd` has the same associations as the struct you are building manually.
Moreover, what is the point of creating a new `card_payment` for this test?
What do _you_ mean by "added new card"? A new card is being added for both the tests above ([L28](https://github.com/aviabird/snitch/blob/card_schema_init/apps/snitch_core/test/data/schema/payment/card_payment_test.exs#L28), [L42](https://github.com/aviabird/snitch/blob/card_schema_init/apps/snitch_core/test/data/schema/payment/card_payment_test.exs#L42))